ts/sagas/startup/__tests__/loadVisibleServicesHandler.test.ts

```diff
@@ -0,0 +1,62 @@
+import { right } from "fp-ts/lib/Either";
+import { testSaga } from "redux-saga-test-plan";
+import { PaginatedServiceTupleCollection } from "../../../../definitions/backend/PaginatedServiceTupleCollection";
+import { ServiceId } from "../../../../definitions/backend/ServiceId";
+import { sessionExpired } from "../../../store/actions/authentication";
+import { loadVisibleServices } from "../../../store/actions/services";
+import { loadVisibleServicesRequestHandler } from "../loadVisibleServicesHandler";
+
+describe("loadVisibleServicesHandler", () => {
+  describe("loadVisibleServicesRequestHandler", () => {
+    const getVisibleServices = jest.fn();
+
+    const mockedVisibleServices: PaginatedServiceTupleCollection["items"] = [
+      { service_id: "A01" as ServiceId, version: 1 },
+      { service_id: "A02" as ServiceId, version: 5 },
+      { service_id: "A03" as ServiceId, version: 2 }
+    ];
+
+    it("returns a generic error if backend response is 500", () => {
+      testSaga(loadVisibleServicesRequestHandler, getVisibleServices)
+        .next()
+        .call(getVisibleServices, {})
+        .next(
+          right({
+            status: 500,
+            value: "An error occurred loading visible services"
+          })
+        )
+        .put(
+          loadVisibleServices.failure(
+            new Error("An error occurred loading visible services")
+          )
+        )
+        .next()
+        .isDone();
+    });
+
+    it("returns proceed to close the sesison on client if backend response is 401", () => {
+      testSaga(loadVisibleServicesRequestHandler, getVisibleServices)
+        .next()
+        .call(getVisibleServices, {})
+        .next(
+          right({
+            status: 401,
+            value: "An error occurred loading visible services"
+          })
+        )
+        .put(sessionExpired())
+        .next()
+        .isDone();
+    });
+
+    it("return an array of visibile services if backend response is 200", () => {
+      testSaga(loadVisibleServicesRequestHandler, getVisibleServices)
+        .next()
+        .call(getVisibleServices, {})
+        .next(right({ status: 200, value: { items: mockedVisibleServices } }))
+        .put(loadVisibleServices.success(mockedVisibleServices))
+        .next();
```

Comment:
```suggestion
       .call(removeOldStoredServices) // removeOldStoredServices
        .call(refreshStoredServices) // refreshStoredServices
        .next()
        .isDone();
```
(you have to export these functions for the test to work)
your refactoring allows to test if some action are dispatched and (if you want) to test these generators too

Reply:
@Undermaken  I opted to create a dedicated store/services folder where store each function on a dedicated file, adding also the code for testing most of the isolated functions. I had problems only with testing the `removeOldStoredServices` function*. just trying with the most simple test such as 
```
import { testSaga } from "redux-saga-test-plan";
import { removeOldStoredServices } from '../removeOldStoredServices';

describe("removeOldStoredServices", () => {

    it("does something", () => {
        testSaga(removeOldStoredServices, {})
        .next()
    })
});
```
it returns the following error:

`TypeError: visibleServices.reduce is not a function`

It seems like the problem is with the reduce method if it is applied to `PaginatedServiceTupleCollection["items"]` objects.

We could proceed by adding the test of the mentioned function with a new pr, having to review the refresh/clean process. Sounds good?

Reply:
@valentinacala  Yes it does!
Track it in a new story please.
Thanks ;)